### PR TITLE
feat: Add Android Clustering

### PR DESCRIPTION
**Pull request by LProulx86**
## Proposed Changes
Add basic clustering to cartography.

## What is the current behavior?
Add clustering on Android. 

## What is the new behavior?
Sample cluster is enabled

## Checklist

Please check if your PR fulfills the following requirements:

- [ ] Documentation has been added/updated
- [ ] Automated Unit / Integration tests for the changes have been added/updated
- [ ] Contains **NO** breaking changes
- [ ] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

File: Samples/Samples/Samples.Shared/DynamicMap_FeaturesPage.xaml [hunk 1]
```diff
@@ -23,6 +23,16 @@
 
             <converters:FromNullableBoolToVisibilityConverter x:Key="IsTrueToCollapsed"
 															  VisibilityIfTrue="Collapsed" />
+            <DataTemplate x:Key="ClusterImageTemplate">
+                <Grid Background="Red"
+                      Width="20"
+                      Height="20">
+                    <TextBlock Text="Bonjour"
```
Comment (carlh98) [inline Samples/Samples/Samples.Shared/DynamicMap_FeaturesPage.xaml:30]: What is the bonjour textblock for?

File: Samples/Samples/Samples.Shared/DynamicMap_FeaturesPage.xaml [hunk 2]
```diff
@@ -112,8 +122,9 @@
                 <dynamicMap:MapControl ViewModel="{Binding}"
 									   SelectionMode="Single"
 									   AutolocateButtonVisibility="Collapsed"
-                                       IsClusterEnabled="{Binding IsClusterEnabled}"
+                                       IsClusterEnabled="{ Binding IsClusterEnabled }"
```
Comment (carlh98) [inline Samples/Samples/Samples.Shared/DynamicMap_FeaturesPage.xaml:125]: remove added spaces

File: Samples/Samples/Samples.Shared/Presentation/DynamicMap_FeaturesPageViewModel.cs [hunk 3]
```diff
@@ -299,17 +299,57 @@ private PushpinEntity[] GetInitialPushpins()
                     new PushpinEntity
                     {
                         Name = "Pushpin 1",
-                        Coordinates = new Geopoint(new BasicGeoposition{Latitude = 46.3938717, Longitude = -72.0921769})
+                        Coordinates = new Geopoint(new BasicGeoposition { Latitude = 45.5040713, Longitude = -73.5587092 })
+                    },
+                    new PushpinEntity
+                    {
+                        Name = "Pushpin 1",
```
Comment (carlh98) [inline Samples/Samples/Samples.Shared/Presentation/DynamicMap_FeaturesPageViewModel.cs:306]: make the names of the pushpins different

File: src/Cartography/Cartography.DynamicMap/Cartography.DynamicMap.csproj [hunk 4]
```diff
@@ -38,6 +38,9 @@
 
 	<ItemGroup Condition="'$(TargetFramework)' == 'xamarinios10'">
 		<PackageReference Include="Xamarin.Google.iOS.Maps" Version="3.9.0" />
+		<PackageReference Include="Xamarin.Google.Maps.Utils">
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Cartography.DynamicMap.csproj:41]: put package reference in one line

File: src/Cartography/Cartography.DynamicMap/Cartography.DynamicMap.csproj [hunk 5]
```diff
@@ -55,6 +58,9 @@
 	</ItemGroup>
 
 	<ItemGroup Condition="'$(TargetFramework)' == 'monoandroid12.0'">
+	  <PackageReference Include="Xamarin.Google.Maps.Utils">
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Cartography.DynamicMap.csproj:61]: same

File: src/Cartography/Cartography.DynamicMap/MapControlBehavior.cs [hunk 6]
```diff
@@ -5,7 +5,9 @@
 using System.Linq;
 using System.Reflection;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices.WindowsRuntime;
 using System.Text;
+
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/MapControlBehavior.cs:10]: remove space

File: src/Cartography/Cartography.DynamicMap/MapControlBehavior.cs [hunk 7]
```diff
@@ -187,7 +196,47 @@ private static void OnPusphinImageSelectorChanged(DependencyObject d, Dependency
...
+                if (d is MapControl map)
+                {
+                    ClusterView.ClusterTemplate = clusterTemplate;
+                }
+#endif
+            }
+#endif
+        }
+
+
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/MapControlBehavior.cs:234]: remove space

File: src/Cartography/Cartography.DynamicMap/MapControlBehavior.cs [hunk 8]
```diff
@@ -187,7 +196,47 @@ private static void OnPusphinImageSelectorChanged(DependencyObject d, Dependency
...
+        public static readonly DependencyProperty ClusterTemplateProperty =
+            DependencyProperty.RegisterAttached("ClusterTemplate", typeof(DataTemplate), typeof(MapControlBehavior), new PropertyMetadata(default, OnClusterTemplateChanged));
+
+
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/MapControlBehavior.cs:238]: remove space

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.cs [hunk 9]
```diff
@@ -0,0 +1,56 @@
+﻿#if __ANDROID__
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.cs:1]: add .Android to file name

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.cs [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+using Android.Widget;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Runtime.Remoting.Contexts;
+using System.Text;
+using Windows.UI.Xaml;
+using Context = Android.Content.Context;
+
+namespace Cartography.DynamicMap.Platforms.Android.Clustering
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.cs:18]: name space should be Cartography.DynamicMap and put the namespace in a single line with ;

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.cs [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+        private DataTemplate _iconTemplate;
+        private DataTemplate _clusterTemplate;
+
+        public ClusterRenderer(DataTemplate iconTemplate, DataTemplate clusterTemplate, Context context, GoogleMap map, ClusterManager clusterManager) : base(context, map, clusterManager)
+        {
+            _iconTemplate = iconTemplate;
+            _clusterTemplate = clusterTemplate;
+        }
+
+        //TODO Add the rendering for customisation of icon and clusterIcon
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.cs:31]: remove the todo section from this PR

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/CustomClusterManager.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+using Android.Content;
+using Android.Gms.Maps;
+using Android.Gms.Maps.Utils.Clustering;
+using Android.Gms.Maps.Utils.Collections;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Cartography.DynamicMap
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/CustomClusterManager.cs:11]: namespace in one line ;

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/MapClusterItem.Android.cs [hunk 13]
```diff
@@ -0,0 +1,32 @@
+﻿#if __ANDROID__
+using Android.Gms.Maps.Model;
+using Android.Gms.Maps.Utils.Clustering;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Cartography.DynamicMap
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/MapClusterItem.Android.cs:8]: namespace one line ;

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/MapClusterItems.Android.cs [hunk 14]
```diff
@@ -0,0 +1,111 @@
+﻿#if __ANDROID__
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Cartography.DynamicMap
+{
+    /// <summary>
+    /// Represent an array of clusterItems
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/MapClusterItems.Android.cs:10]: put a dot a the end of sentence

File: Samples/Samples/Samples.Shared/DynamicMap_FeaturesPage.xaml (hunk 1 above)
Comment (LProulx86) [reply Samples/Samples/Samples.Shared/DynamicMap_FeaturesPage.xaml:30]: Testing DataTemplate for styling. I can Remove it !

File: src/Cartography/Cartography.DynamicMap/Cartography.DynamicMap.csproj [hunk 15]
```diff
@@ -38,9 +38,7 @@
 
 	<ItemGroup Condition="'$(TargetFramework)' == 'xamarinios10'">
 		<PackageReference Include="Xamarin.Google.iOS.Maps" Version="3.9.0" />
-		<PackageReference Include="Xamarin.Google.Maps.Utils">
-		  <Version>1.3.4</Version>
-		</PackageReference>
+		<PackageReference Include="Xamarin.Google.Maps.Utils"><Version>1.3.4</Version></PackageReference>
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Cartography.DynamicMap.csproj:41]: Should be <PackageReference Include="Xamarin.Google.Maps.Utils" Version="1.3.4" />

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.Android.cs [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+using Android.Widget;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Runtime.Remoting.Contexts;
+using System.Text;
+using Windows.UI.Xaml;
+using Context = Android.Content.Context;
+
+namespace Cartography.DynamicMap;
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.Android.cs:18]: add space

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.Android.cs [hunk 17]
```diff
@@ -0,0 +1,29 @@
+﻿#if __ANDROID__
+using Android.Content;
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.Android.cs:2]: reorder usings

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/MapClusterItems.Android.cs [hunk 18]
```diff
@@ -0,0 +1,110 @@
+﻿#if __ANDROID__
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Cartography.DynamicMap;
+
+/// <summary>
+/// Represent an array of clusterItems
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/MapClusterItems.Android.cs:10]: add dot at the end

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/MapClusterItems.Android.cs [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+/// </summary>
+public class MapClusterItems
+{
+
+    /// <summary>
+    /// Initiate the array of cluster Items.
+    /// </summary>
+    /// <param name="mapClusterItems"></param>
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/MapClusterItems.Android.cs:18]: add dot at the end for param also

File: src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.Android.cs [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+using System.Runtime.Remoting.Contexts;
+using System.Text;
+using Windows.UI.Xaml;
+using Context = Android.Content.Context;
+
+namespace Cartography.DynamicMap;
+internal class ClusterRenderer : DefaultClusterRenderer
+{
+    private DataTemplate _iconTemplate;
+    private DataTemplate _clusterTemplate
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/ClusterRenderer.Android.cs]: missing ; at the end
also add space after

File: src/Cartography/Cartography.DynamicMap/Platforms/iOS/ClusterView.iOS.cs [hunk 21]
```diff
@@ -0,0 +1,92 @@
+﻿#if __IOS__
```
Comment (carlh98) [inline src/Cartography/Cartography.DynamicMap/Platforms/iOS/ClusterView.iOS.cs:1]: remove ios from your PR

**Pull request closed without merging**